verilog/analysis/verilog_project.cc

```diff
@@ -345,4 +347,59 @@ absl::StatusOr<FileList> ParseSourceFileListFromFile(
   return ParseSourceFileList(file_list_file, content);
 }
 
+absl::StatusOr<FileList> ParseSourceFileListFromCommandline(
+    const std::vector<absl::string_view>& cmdline) {
+  FileList result;
+  for (absl::string_view argument : cmdline) {
+    if (argument.empty()) continue;
+    if (argument[0] != '+') {
+      // Then "argument" is a SV file name.
+      result.file_paths.push_back(std::string(argument));
+      continue;
+    }
+    // It should be either a define or incdir.
+    std::vector<absl::string_view> argument_plus_splitted =
+        absl::StrSplit(argument, absl::ByChar('+'), absl::SkipEmpty());
+    if (argument_plus_splitted.size() < 2) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("ERROR: Expected either '+define+' or '+incdir+' "
+                       "followed by the parameter but got '",
+                       argument, "'"));
+    }
+    absl::string_view plus_argument_type = argument_plus_splitted[0];
+    if (plus_argument_type == "define") {
+      // define argument.
```

Comment:
To reduce visual noise, I'd probably leave out this comment as it more or less repeats what the code line says.
(similar below for incdir).